### ISSUE TITLE
With "Override local DNS" set and using an exit node, all DNS resolution fails on Linux

### ISSUE THREAD

**Issue by darshinimashar** [tailscale/tailscale#3060]
Internet does not work when we enable the Override local DNS with exitnode is enabled 

![image](https://user-images.githubusercontent.com/75628398/137024225-6a174794-5430-43b6-9fd5-b391c11f6cb4.png)

Tested on Ubuntu 20.04

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_3nump)

**Comment by bradfitz** [comment]
I can reproduce getting into a broken state. Info, not sure how much is relevant yet:

* Ubuntu 20.04 desktop
* On a LAN with IPv6
* Tailscale 1.16.0
* Tailnet DNS: 8.8.8.8 + override enabled

Everything works normally until I use an exit node, and then:

```
root@ub2004desk:/home/bradfitz# tailscale up --exit-node=100.78.238.67
root@ub2004desk:/home/bradfitz# tailscale status
# Health check:
#     - router: 2 add route failures; first was: running "ip route add 2602:d1:xxx::/64 dev tailscale0 table 52" failed: exit status 2
RTNETLINK answers: File exists

```

I can ping 100.100.100.100, but not 8.8.8.8 then.

Routes at least made it into Tailscale:

```
# ip route show table all
default dev tailscale0 table 52 scope link
10.0.0.0/16 dev tailscale0 table 52 scope link
100.74.218.60 dev tailscale0 table 52 scope link
100.75.80.117 dev tailscale0 table 52 scope link
100.78.238.67 dev tailscale0 table 52 scope link
...
```

But then I imagine due to the error above, it didn't finish configuring the host router/etc settings.

**Comment by bradfitz** [comment]
There's something going wrong with the conversion from the `ExitNodeIP` to `ExitNodeID` pref in tailscaled.

When an exit node eventually finally somehow works, it looks like this:

```
root@ub2004desk:/home/bradfitz# ./tailscale debug -prefs  | grep Exit
	"ExitNodeID": "n7TLwD5CNTRL",
	"ExitNodeIP": "",
	"ExitNodeAllowLANAccess": false,
root@ub2004desk:/home/bradfitz# ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=58 time=42.1 ms
^C
--- 8.8.8.8 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 42.148/42.148/42.148/0.000 ms
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
root@ub2004desk:/home/bradfitz# ping google.com
PING google.com (142.251.33.110) 56(84) bytes of data.
64 bytes from sea30s10-in-f14.1e100.net (142.251.33.110): icmp_seq=1 ttl=58 time=45.5 ms
64 bytes from sea30s10-in-f14.1e100.net (142.251.33.110): icmp_seq=2 ttl=58 time=36.0 ms
^C
--- google.com ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 36.040/40.764/45.488/4.724 ms
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
root@ub2004desk:/home/bradfitz# curl https://canhazip.com/
24.16.10.156
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
```

Then I can turn it off, and that works too:

```
root@ub2004desk:/home/bradfitz# ./tailscale up --exit-node=
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
2602:xxxx:d735:65a0:8200:b498
```

But once I turn it right back on, to the same exit node I was just successfully using:

```
root@ub2004desk:/home/bradfitz# ./tailscale up --exit-node=100.78.238.67 
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/

^C
root@ub2004desk:/home/bradfitz# ./tailscale debug -prefs  | grep Exit
	"ExitNodeID": "",
	"ExitNodeIP": "100.78.238.67",
	"ExitNodeAllowLANAccess": false,
```

And all network is busted.

But some time passes, and:

```
root@ub2004desk:/home/bradfitz# tailscale debug -prefs  | grep Exit
Warning: client version "1.16.0-t86bcf6a96-gf787a582e" != tailscaled server version "date.20211004"
	"ExitNodeID": "n7TLwD5CNTRL",
	"ExitNodeIP": "",
	"ExitNodeAllowLANAccess": false,
root@ub2004desk:/home/bradfitz# ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=58 time=36.0 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=58 time=32.8 ms
^C
--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 32.836/34.401/35.966/1.565 ms
root@ub2004desk:/home/bradfitz# curl http://canhazip.com/
24.16.10.156
```

... it's back to working.

**Comment by pkel** [comment]
A quick workaround is to `sudo systemctl restart tailscaled.service` after setting the exit node.

Before the restart, network is broken and `sudo tailscale debug -prefs  | grep Exit` shows the IP but not the ID. After the restart the network is ok and  `sudo tailscale debug -prefs  | grep Exit` shows the ID not the IP.

```
$ sudo tailscale version
1.16.2
  tailscale commit: f449111792f662c8552514cf3691c3c1bd7f38e4
  other commit: 3853a3b920c7ec33da2b8ba8454e9ca8b3772774
  go version: go1.17.1-ts097d1284f9
```

**Comment by DentonGentry** [comment]
This is expected to be resolved for keepsies by fixing #1713 in the 1.20 release in January.

**Comment by DentonGentry** [comment]
This specific issue was resolved in 1.20 by sending DNS queries to the exit node.
There have been a couple issues in 1.20 but these are being tracked in https://github.com/tailscale/tailscale/issues/3842

Closing this one as fixed.
